cmd/bootstrap/cmd/observer_network_key.go

```diff
@@ -0,0 +1,78 @@
+package cmd
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/flow-go/cmd"
+	"github.com/onflow/flow-go/cmd/bootstrap/utils"
+	"github.com/onflow/flow-go/crypto"
+)
+
+var (
+	flagOutputFile string
+)
+
+// observerNetworkKeyCmd represents the `observer-network-key` command which generates required network key
+// for an Observer,  and writes it to the default path within the provided directory. Used by new Observer
+// operators to create the networking key only
+var observerNetworkKeyCmd = &cobra.Command{
+	Use:   "observer-network-key",
+	Short: "Generates network key and writes it to the default path within the output directory",
+	Run:   observerNetworkKeyRun,
+}
+
+func init() {
+	rootCmd.AddCommand(observerNetworkKeyCmd)
+
+	observerNetworkKeyCmd.Flags().StringVarP(&flagOutputFile, "output-file", "f", "", "output file path")
+	cmd.MarkFlagRequired(observerNetworkKeyCmd, "output-file")
+
+	observerNetworkKeyCmd.Flags().BytesHexVar(
+		&flagNetworkSeed,
+		"seed",
+		[]byte{},
+		fmt.Sprintf("hex encoded networking seed (min %d bytes)", crypto.KeyGenSeedMinLenECDSASecp256k1))
+}
+
+// observerNetworkKeyRun generate a network key and writes it to a default file path.
+func observerNetworkKeyRun(_ *cobra.Command, _ []string) {
+
+	// generate seed if not specified via flag
+	if len(flagNetworkSeed) == 0 {
+		flagNetworkSeed = GenerateRandomSeed(crypto.KeyGenSeedMinLenECDSASecp256k1)
+	}
+
+	// if the file already exists, exit
+	keyExists, err := pathExists(flagOutputFile)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not check if %s exists", flagOutputFile)
+	}
+
+	if keyExists {
+		log.Warn().Msgf("%s already exists, exiting...", flagOutputFile)
+		return
+	}
+
+	// generate unstaked networking private key
+	networkKey, err := utils.GenerateUnstakedNetworkingKey(flagNetworkSeed)
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not generate network key")
+	}
+	log.Info().Msg("generated network key")
+
+	// hex encode and write to file
+	output := make([]byte, hex.EncodedLen(networkKey.Size()))
```

Comment:
Is `.Size()` returning encoded length? 
I'm not sure if we need really to do this pre-allocation, can't we just encode whatever we get?
This might lead to some subtle bugs in the future

Reply:
fair enough. I updated to use the actual slice len. the preallocation is needed. that's actually what the [hex library does](https://cs.opensource.google/go/go/+/master:src/encoding/hex/hex.go;l=106-109?q=hex.go&ss=go%2Fgo:src%2F) for `EncodeToString` except without the cast